scandium-core/src/main/java/org/eclipse/californium/scandium/dtls/InMemoryConnectionStore.java

```diff
@@ -175,12 +204,31 @@ public void onEviction(Connection staleConnection) {
 	 * 
 	 * @param tag tag for logging
 	 */
-	public synchronized void setTag(final String tag) {
-		if (tag.endsWith(" ")) {
+	public synchronized InMemoryConnectionStore setTag(final String tag) {
+		if (tag.isEmpty() || tag.endsWith(" ")) {
 			this.tag = tag;
 		} else {
 			this.tag = tag + " ";
 		}
+		return this;
+	}
+
+	/**
+	 * Creates a new unused connection id.
+	 * 
+	 * @return connection id, or {@code null}, if no free connection id could
+	 *         created
+	 */
+	private ConnectionId newConnectionId() {
+		byte[] cidBytes = new byte[cidLength];
+		for (int i = 0; i < 10; ++i) {
+			random.nextBytes(cidBytes);
+			ConnectionId cid = new ConnectionId(cidBytes);
+			if (connections.get(cid) == null) {
+				return cid;
+			}
```

Comment:
The idea is to use a cid length, which is at least 2 bytes larger then the expected connections. For this commit the DEFAULT_CID_LENGTH is just a short cut for the later introduction of the configurable value.
According your comment above, I added a info message about dropping a session ticket. FMPOV info is enough. If someone resumes, it will just fall into a full-handshake.